### DISCUSSION TITLE
Feature Request: support hf:// protocol in wasm httpfs

### DISCUSSION THREAD

**Discussion by cfahlgren1** [duckdb/duckdb-wasm#1858] [Ideas]
The `hf://` protocol is supported in native DuckDB ([Docs](https://duckdb.org/2024/05/29/access-150k-plus-datasets-from-hugging-face-with-duckdb.html)). Creating this to track progress with porting to DuckDB WASM.

[Relevant Comment in Discord
](https://discord.com/channels/909674491309850675/921100929984503858/1283690700206112783)

**Comment by severo** [comment]
Working now!

```
SET builtin_httpfs = true; LOAD httpfs;
```

https://www.npmjs.com/package/@duckdb/duckdb-wasm/v/1.31.0

<img width="695" height="334" alt="Screenshot From 2025-09-29 16-59-44" src="https://github.com/user-attachments/assets/a9cdf9d1-b920-4cf2-a175-9355860f17a1" />

[console](https://shell.duckdb.org/#queries=v0,SET-builtin_httpfs-%3D-true~-LOAD-httpfs~,select-count(*)-from-'hf%3A%2F%2Fdatasets%2Fcfahlgren1%2Fhub%20stats%2Fdatasets.parquet'~)

---

Will `builtin_httpfs` become the default soon, @carlopi?

**Reply by carlopi** [reply]
Yes, only remaining blocker is deleting existing code, aim is together with 1.4.1 (presumed next week Monday).

Basically at the moment there are 2 HTTPFS implementation, but only the one provided by the extensions should actually be there.
